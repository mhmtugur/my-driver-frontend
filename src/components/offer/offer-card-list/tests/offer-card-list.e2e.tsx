import { E2EElement, E2EPage, newE2EPage } from '@stencil/core/testing';

describe('offer card list list', () => {
  let page: E2EPage;
  let element: E2EElement;
  const offerList = [
    {
      web_img: 'car.png',
      amount: 200,
      title: 'Mercedes E200',
      exampleCar: 'Example Car',
      currency: 'EUR',
    },
    {
      web_img: 'car.png',
      amount: 300,
      title: 'BMW 7',
      exampleCar: 'Example Car',
      currency: 'EUR',
    },

  ]
  beforeEach(async () => {
    page = await newE2EPage({ html: '<offer-card-list></offer-card-list>' });
    element = await page.find('offer-card-list');
  });

  it('should render offer list', async () => {
    element.setProperty('offerList', offerList);
    await page.waitForChanges();
    const cardList = await page.findAll('.card');
    expect(cardList.length).toBe(2);
    expect(element.innerHTML).toMatchSnapshot();
  });
});