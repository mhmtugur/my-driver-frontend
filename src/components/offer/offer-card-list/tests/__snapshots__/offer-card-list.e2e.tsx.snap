// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`offer card list list should render offer list 1`] = `"<div><div class=\\"col s12 m6 l4\\"><offer-card class=\\"hydrated\\"><div class=\\"card\\"><div class=\\"card-image\\"><img src=\\"car.png\\"></div><div class=\\"card-content\\"><p class=\\"card-right-text\\">200 EUR</p><br><br><p class=\\"card-title truncate\\">Mercedes E200</p><p class=\\"truncate\\">Example Car</p></div><div class=\\"card-action\\"><a class=\\"btn waves-effect waves-light\\">SELECT</a></div></div></offer-card></div><div class=\\"col s12 m6 l4\\"><offer-card class=\\"hydrated\\"><div class=\\"card\\"><div class=\\"card-image\\"><img src=\\"car.png\\"></div><div class=\\"card-content\\"><p class=\\"card-right-text\\">300 EUR</p><br><br><p class=\\"card-title truncate\\">BMW 7</p><p class=\\"truncate\\">Example Car</p></div><div class=\\"card-action\\"><a class=\\"btn waves-effect waves-light\\">SELECT</a></div></div></offer-card></div></div>"`;
